pkg/apis/v1beta1/taints.go

```diff
@@ -0,0 +1,33 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import v1 "k8s.io/api/core/v1"
+
+// Karpenter specific taints
+const (
+	DisruptionTaintKey             = Group + "/disruption"
+	DisruptingNoScheduleTaintValue = "disrupting"
```

Comment:
So, I'd call the taint “node-consolidation-planned”. It's tricky though because if we have lots of taints and people want to tolerate them, they have to learn what those taints are (or tolerate `*`, which has its own problems).

The actual name isn't the point here; it's the idea of having a taint that's specific to each different intent.
If future Karpenter drains nodes due to an elevated spot termination risk, then IMO the taint could and should be different.

Reply:
@sftim Can you detail what the problems are with tolerating `*`? The original intent around making this a single taint key with multiple values was to make it easier for users to tolerate all of the Karpenter taints across the board without having to think about it too much. It seems like the node SIG takes the opposite approach where each part of the lifecycle has its own taint. I'm curious what the strengths of drawbacks of each of these approaches are in your opinion.

Reply:
The trouble with tolerations is that you can only tolerate `*`. So if there's a taint `example.com/node-actually-on-fire` and you tolerate `*`, your Pod schedules there.

What you probably want to do is to tolerate families of taints. You might have a DaemonSet pod that provides packet filtering (and fails closed).

What you want is roughly to deploy first, drain last, and tolerate at least:
- `node.kubernetes.io/memory-pressure`
- `node.kubernetes.io/disk-pressure`
- `node.kubernetes.io/pid-pressure`
- `node.kubernetes.io/network-unavailable`

(which, fortunately, the DaemonSet controller adds automatically).

You've never heard of `spot-compute.example/preemption-imminent` and you'd like to tolerate that. The problem is, you don't want to tolerate _every_ taint.

Anyway, the risk of having lots of fine grained taints is grumpy sysadmins moaning that there's a growing list of taints they need to explicitly tolerate. The answer may be to expect a level of moaning and have clear documentation to steer people towards.

Reply:
If we plan to add future taints, we can reserve them and document the reservation. That helps people plan ahead and tolerate the taints now, even before we ever use them.

Reply:
> The trouble with tolerations is that you can only tolerate *.

I just tried this out on my cluster and saw tolerations apply with specific non * values. 

I can understand the grumpy sysadmins case. We plan on adding this taint in, but I think the only other two taints would be mutually exclusive with this taint and definitely fully documented when we progress on those.

Reply:
Oh, yeah, that was unclear. I meant that `*` is the only wildcard available.